VideoApp/Video-TwilioUITests/Activities/AuthActivities.swift

```diff
@@ -35,6 +35,8 @@ class AuthActivities {
             passwordTextField.typeText(password)
 
             app.buttons["submitButton"].tap()
+            
+            sleep(3) // Have to wait for multiple screen transitions to complete
```

Comment:
I think when we make sign in UI design changes we will end up with a single screen transition after sign in so UI tests should just work and we can remove this.

Reply:
Maybe we need a quick chat on this. I am not sure what is requiring the current sleep call?

Reply:
I believe we should be able to write UI tests by adding commands for button taps and such across multiple screens and not have to use sleep and it all just works. I think the test framework is supposed to wait for the transition between screens to complete before resuming actions.

But after sign in completes we've got some pretty complicated navigation happening that I think is causing problems with the test framework. We are doing 3 screen transitions back to back. 
```swift
            navigationVC.dismiss(animated: true) {
                navigationVC.popToRootViewController(animated: true)
                navigationVC.viewControllers.first?.performSegue(withIdentifier: "lobbySegue", sender: self)
            }
```
Adding a short sleep after sign in allows all of these transitions to complete before the next test commands run. It's not awesome but not really a bad problem right now since we only have a few UI tests.

I think when we implement sign in UI improvements this should just go away. The new design that we have a wireframe for would result in a single dismiss to transition from sign in to lobby. This also requires some reorganization of how the UI stack is created at launch.

Reply:
Shoot CI is consistently failing after removing the sleep. So I replaced the sleep with an async expect which is better.

We need to understand why this and the intermittent deep link failure are problems on CI but never locally. Maybe CI has slower machines and there is some race condition. But I think this is an effort by itself to figure it out and the status check must be reliable.

There are also some UI design/architecture things that I think may not be helping. Sign in doesn't display any loader during the network request so we can't really wait until that finishes to proceed with test. And when we transition from sign in to lobby we are doing 3 back to back screen transitions.
```swift
            navigationVC.dismiss(animated: true) {
                navigationVC.popToRootViewController(animated: true)
                navigationVC.viewControllers.first?.performSegue(withIdentifier: "lobbySegue", sender: self)
            }
```
We have a new sign in UI designed and I think when we implement it these things will be addressed. But not sure if that will fix CI.